### ISSUE TITLE
Spaces in dataset name leads to wrong URL on `build`

### ISSUE THREAD

**Issue by DavidS** [honeycombio/buildevents#72]
The URL printed by `buildevents build` translates spaces in the dataset name to `%20`, while the honeycomb UI uses dashes. So a dataset called `litmus tests` should be linked as `litmus-tests` but is linked as `litmus%20tests`, which does not work.

**Comment by kvrhdn** [comment]
Hi, saw this pass by on Slack.

While working on the Terraform provider, I encountered a similar issue but with `/`. Now I'm wondering what all the characters are that get replaced by the API... 🤔 Maybe someone from Honeycomb can give a definitive list?

I've added the following utility function myself:
```go
func urlEncodeDataset(dataset string) string {
	return strings.Replace(dataset, "/", "-", -1)
}
```
https://github.com/kvrhdn/go-honeycombio/blob/main/client.go#L185:L187

This will change a dataset named `kvrhdn/go-honeycombio` to `kvrhdn-go-honeycombio`.
